File: src/main/java/wordcount/ui/ConsoleInputReader.java
package wordcount.ui;

import java.util.Scanner;

public class ConsoleInputReader implements UserInputReader {
	protected static final String INPUT_PROMPT = "Enter text: ";

	@Override
	public String readUserInput() {
		// 1. prompt
		System.out.print(INPUT_PROMPT);
		// 2. read input
		String line;
		try (Scanner scanner = new Scanner(System.in)) {
			line = scanner.nextLine();
		}
		return line;
	}
}

File: src/main/kotlin/inputReader/ConsoleInputReader.kt
package inputReader

class ConsoleInputReader(private val readLineImplementation: () -> String, private val printerImplementation: (String) -> Unit): InputReader {
    companion object {
        const val entryMessage = "Enter text: "
    }
    override fun read(): String {
        printerImplementation(entryMessage)
        return readLineImplementation()
    }
}